rotkehlchen/tests/api/test_bitcoin.py

```diff
@@ -270,10 +283,34 @@ def test_add_delete_xpub(rotkehlchen_api_server):
     else:
         assert_proper_response(response)
 
-    # test that adding the same BCH xpub for BTC works
+    # Check that periodic derivation doesn't break anything
+    # Testing here since test_tasks_manager.py only tests scheduling
+    rotki.task_manager.last_xpub_derivation_ts = 0   # to be sure that the task will be scheduled
+    rotki.task_manager._maybe_schedule_xpub_derivation()
+    while len(rotki.task_manager.greenlet_manager.greenlets) != 0:
```

Comment:
you should put this for loop in a `with gevent.Timeout(ASYNC_TASK_WAIT_TIMEOUT):`. So that if for some reason it does not work then the test fails after the timeout, otherwise the test will wait forever